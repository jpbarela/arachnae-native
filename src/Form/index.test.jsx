// @flow
import * as React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import { NumericInput } from "./index";

describe("NumericInput", () => {
  test("renders", () => {
    const container = render(<NumericInput />);
    expect(container).toMatchSnapshot();
  });

  test("returns a numeric input", () => {
    const changeHandler = jest.fn();
    const placeholderText = "Test numeric conversion";

    const { getByPlaceholderText } = render(
      <NumericInput placeholder={placeholderText} onChange={changeHandler} />
    );

    const input = getByPlaceholderText(placeholderText);
    fireEvent(input, "onEndEditing", { nativeEvent: { text: "23" } });

    expect(changeHandler.mock.calls[0][0]).toBe(23);
  });
});
