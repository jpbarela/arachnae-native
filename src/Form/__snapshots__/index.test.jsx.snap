// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericInput renders 1`] = `
<TextInput
  keyboardType="numeric"
  onEndEditing={[Function]}
  selectionColor="black"
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "arial",
        "fontSize": 28,
      },
      Object {},
    ]
  }
/>
`;
